lib/internal/dns/compat.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+
+const {
+  ArrayPrototypeFilter,
+  ArrayPrototypeMap,
+  ArrayPrototypeSort,
+  JSONParse,
+} = primordials;
+
+const { DNSWrap } = internalBinding('dns');
+const { dnsException } = require('internal/errors');
+const { AsyncResource } = require('async_hooks');
+
+class ReqWrap {
+  #resource;
+  #oncomplete;
```

Comment:
Hidden symbols are still quite a bit faster than private fields. For now, I'd rather avoid the private fields.